moshi/src/test/java/com/squareup/moshi/ObjectAdapterTest.java

```diff
@@ -84,6 +85,17 @@
     assertThat(adapter.fromJson("[0, 1]")).isEqualTo(Arrays.asList(0d, 1d));
   }
 
+  @Test public void fromJsonDoesNotFailOnNullValues() throws Exception {
+    Moshi moshi = new Moshi.Builder().build();
+    JsonAdapter<Object> adapter = moshi.adapter(Object.class);
+
+    try {
+      adapter.fromJson("{\"address\":null, \"items\":null}");
```

Comment:
nit: the try/catch isn't really necessary. Better would be to confirm that the result matches a map with two keys whose values are null